### PR TITLE
Update README to add submodule preview workflow

### PR DESCRIPTION
**Pull request by domlet**
@danswick Would it be okay to add these instructions (to preview changes to the mapbox-gl-js submodule) here?

Thanks for your help and your 👀 !

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,6 +73,8 @@ The command will print the URL you can use to view the documentation.
 
 💡 If you receive an error related to `@mapbox/appropriate-images`, try `nvm use && npm start`.
 
+💡 You can use this repo to preview changes to content in the `mapbox-gl-js` submodule. To do this, make your changes to files in the `node_modules` folder, save the files, and then use `npm start` to start the server. You must make your *actual* changes in the [mapbox-gl-js](https://github.com/mapbox/mapbox-gl-js) repo, but you can use this process to preview your edits in a web browser. You will not be able to commit your changes to files in the `node_modules` folder.
```
Comment (danswick) [inline README.md:76]: I _think_ this only applies to changes to the style spec source code. For changes to the main mapbox-gl-js source code, we could summarize some of the content from the old examples sprint plan: https://github.com/mapbox/documentation/issues/403#issuecomment-612217143. That might be a bit much, but perhaps we should err on the side of over-documenting? What do you think, @domlet?

**Review by danswick — COMMENTED** [review]
Thanks so much for documenting this, @domlet! I left a comment about distinguishing between the style spec and main gl js source code. lmk what you think.

File: README.md (hunk 1 above)
Comment (domlet) [reply README.md]: Thank you @danswick – I've taken a moment to test the steps for previewing changes to `gl js` source code files, and added a new section covering that: https://github.com/mapbox/mapbox-gl-js-docs/pull/420/commits/ac7cbb5e70786bd90949b2a4440b315afae5b0c8 

Let me know if this looks good to you!

**Review by danswick — APPROVED** [review]
@domlet I caught this branch up with publisher-production just to make sure everything is copacetic. Looks good!

**Pull request merged**